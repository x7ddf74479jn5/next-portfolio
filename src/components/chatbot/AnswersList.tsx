import React from "react";
import styles from "src/styles/components/chatbot/AnswerList.module.scss";
import type { SelectAnswer } from "src/types/chatbot";
import type { Answer as AnswerType } from "src/types/chatbot";

import Answer from "./Answer";

type Props = {
  answers: AnswerType[];
  select: SelectAnswer;
};

const AnswersList: React.VFC<Props> = (props) => {
  return (
    <div className={styles.gridAnswer}>
      {props.answers.map((value, index) => {
        return <Answer content={value.content} nextId={value.nextId} key={index.toString()} select={props.select} />;
      })}
    </div>
  );
};

export default AnswersList;
