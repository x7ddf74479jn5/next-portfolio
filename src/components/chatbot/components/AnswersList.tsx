import React from "react";
import type { SelectAnswer } from "src/types/chatbot";

import { Answer } from "./index";

type Props = {
  answers: Answer[];
  select: SelectAnswer;
};

const AnswersList: React.VFC<Props> = (props) => {
  return (
    <div className={"c-grid__answer"}>
      {props.answers.map((value, index) => {
        return <Answer content={value.content} nextId={value.nextId} key={index.toString()} select={props.select} />;
      })}
    </div>
  );
};

export default AnswersList;
