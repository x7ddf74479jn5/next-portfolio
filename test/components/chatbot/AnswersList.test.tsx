import AnswersList from "src/components/chatbot/AnswersList";
import type { Answer } from "src/types/chatbot";
import type { RenderResult } from "test/test-utils";
import { render } from "test/test-utils";

describe("AnswersList", () => {
  let renderResult: RenderResult;
  const mockHandleSelect = jest.fn();
  const mockHandleIsSelectReady = jest.fn().mockReturnValue(true);

  const testData: Answer[] = [
    {
      content: "content",
      nextId: "nextId",
    },
    {
      content: "content 2",
      nextId: "nextId 2",
    },
  ];

  beforeEach(() => {
    renderResult = render(
      <AnswersList answers={testData} select={mockHandleSelect} isSelectReady={mockHandleIsSelectReady} />
    );
  });

  it("matches snapshot", () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    expect(renderResult.container.firstElementChild).toHaveClass("gridAnswer");
    expect(renderResult.getAllByRole("button")).toHaveLength(2);
  });
});
