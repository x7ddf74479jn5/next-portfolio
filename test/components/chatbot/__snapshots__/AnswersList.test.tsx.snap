// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswersList matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="gridAnswer"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-1"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        content
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-1"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        content 2
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
